### PR TITLE
[Enhancement] Optimize pose tracker

### PR DESCRIPTION
**Pull request by lzhangzz**
## Modification

1. Add WarpAffine operation with CPU/CUDA support
2. Add a preload transform to avoid frequent HtoD copy
3. Add OKS-based NMS to avoid overlapping tracks
4. Add One-euro filter to smooth tracks
5. Merge detection bboxes and tracked bboxes in detection frame to avoid detector FN
6. Add 2-stage assignment to improve tracking performance
7. Add key-point threshold
8. Add min bbox size threshold
9. Add max num bbox option
10. Add better visualization
11. Add missing track recovery
12. Re-organize code

### CODE REVIEW

File: demo/csrc/cpp/pose_tracker.cpp [hunk 1]
```diff
@@ -15,9 +17,17 @@
 const auto config_json = R"(
 {
   "type": "Pipeline",
-  "input": ["data", "use_det", "state"],
+  "input": ["img", "use_det", "state"],
   "output": "targets",
   "tasks": [
+    {
+      "type": "Task",
```
Comment (lvhan028) [inline demo/csrc/cpp/pose_tracker.cpp:24]: nice

**Comment by lvhan028** [comment]
@Tau-J Could you help review this PR?

File: demo/csrc/cpp/pose_tracker.cpp [hunk 2]
```diff
@@ -92,223 +115,781 @@ std::optional<std::array<float, 4>> keypoints_to_bbox(const std::vector<cv::Poin
...
-Value ProcessBboxes(const Value& detections, const Value& data, Value state) {
-  assert(state.is_pointer());
-  Value::Array bboxes;
-  if (detections.is_array()) {  // has detections
-    auto& dets = detections.array();
-    for (const auto& det : dets) {
-      if (det["label_id"].get<int>() == 0 && det["score"].get<float>() >= .3f) {
-        bboxes.push_back(det);
+  // TopDownAffine's internal logic for mapping pose detector inputs
+  Bbox MapBbox(const Bbox& box) {
```
Comment (lvhan028) [inline demo/csrc/cpp/pose_tracker.cpp:328]: Should it be kept? `MapBbox` is not used.

**Review by lvhan028 — APPROVED** [review]
LGTM

**Review by irexyc — APPROVED** [review]

**Pull request merged**